src/types/effects.ts

```diff
@@ -0,0 +1,251 @@
+  claimable_balance_clawed_back = 80,
+}
+export interface BaseEffectRecord extends Horizon.BaseResponse {
+  id: string;
+  account: string;
+  paging_token: string;
+  type_i: EffectType;
+  type: string;
+  created_at: string;
+}
+export interface AccountCreated extends BaseEffectRecord {
+  type_i: EffectType.account_created;
+  starting_balance: string;
+}
+export interface AccountCredited extends BaseEffectRecord, OfferAsset {
+  type_i: EffectType.account_credited;
+  amount: string;
+}
+export interface AccountDebited extends BaseEffectRecord {
+  type_i: EffectType.account_debited;
+  amount: string;
+}
+export interface AccountThresholdsUpdated extends BaseEffectRecord {
+  type_i: EffectType.account_thresholds_updated;
+  low_threshold: number;
+  med_threshold: number;
+  high_threshold: number;
+}
+export interface AccountHomeDomainUpdated extends BaseEffectRecord {
+  type_i: EffectType.account_home_domain_updated;
+  home_domain: string;
+}
+export interface AccountFlagsUpdated extends BaseEffectRecord {
+  type_i: EffectType.account_flags_updated;
+  auth_required_flag: boolean;
+  auth_revokable_flag: boolean;
+}
+interface DataEvents extends BaseEffectRecord {
+  name: boolean;
+  value: boolean;
+}
+export interface DataCreated extends DataEvents {
+  type_i: EffectType.data_created;
+}
+export interface DataUpdated extends DataEvents {
+  type_i: EffectType.data_updated;
+}
+export interface DataRemoved extends DataEvents {
+  type_i: EffectType.data_removed;
+}
+export interface SequenceBumped extends BaseEffectRecord {
+  type_i: EffectType.sequence_bumped;
+  new_seq: number | string;
+}
+interface SignerEvents extends BaseEffectRecord {
+  weight: number;
+  key: string;
+  public_key: string;
+}
+export interface SignerCreated extends SignerEvents {
+  type_i: EffectType.signer_created;
+}
+export interface SignerRemoved extends SignerEvents {
+  type_i: EffectType.signer_removed;
+}
+export interface SignerUpdated extends SignerEvents {
+  type_i: EffectType.signer_updated;
+}
+interface TrustlineEvents extends BaseEffectRecord, OfferAsset {
+  limit: string;
+}
+export interface TrustlineCreated extends TrustlineEvents {
+  type_i: EffectType.trustline_created;
+}
+export interface TrustlineRemoved extends TrustlineEvents {
+  type_i: EffectType.trustline_removed;
+}
+export interface TrustlineUpdated extends TrustlineEvents {
+  type_i: EffectType.trustline_updated;
+}
+export interface TrustlineAuthorized extends BaseEffectRecord {
+  type_i: EffectType.trustline_authorized;
+  asset_type: OfferAsset["asset_type"];
+  asset_code: OfferAsset["asset_code"];
+  trustor: string;
+}
+export interface TrustlineDeautorized
+  extends Omit<TrustlineAuthorized, "type_i"> {
+  type_i: EffectType.trustline_deauthorized;
+}
+export interface TrustlineAutorizedToMaintainLiabilities
+  extends Omit<TrustlineAuthorized, "type_i"> {
+  type_i: EffectType.trustline_authorized_to_maintain_liabilities;
+}
```

Comment:
Ahh looks like one last typo: TrustlineAutorized -> TrustlineAut**h**orized here

Reply:
Good catch too, I will review the whole file just to make sure 👌

Reply:
Caught the deauthorized one too with a typo, this should be good now.